---a/net/minecraft/entity/monster/EntitySpider.java
+++b/net/minecraft/entity/monster/EntitySpider.java
@@ -1,6 +1,7 @@
 package net.minecraft.entity.monster;
 
 import java.util.Random;
+
 import net.minecraft.block.Block;
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.entity.EnumCreatureAttribute;
@@ -26,12 +26,12 @@
 import net.minecraft.world.EnumDifficulty;
 import net.minecraft.world.World;
 
-public class EntitySpider extends EntityMob
-{
+public class EntitySpider extends EntityMob {
+
     private static final String __OBFID = "CL_00001699";
 
-    public EntitySpider(World worldIn)
-    {
+    public EntitySpider(World worldIn) {
         super(worldIn);
         this.setSize(1.4F, 0.9F);
         this.tasks.addTask(1, new EntityAISwimming(this));
@@ -47,220 +47,176 @@
         this.targetTasks.addTask(3, new EntitySpider.AISpiderTarget(EntityIronGolem.class));
     }
 
-    protected PathNavigate func_175447_b(World worldIn)
-    {
+    protected PathNavigate func_175447_b(World worldIn) {
         return new PathNavigateClimber(this, worldIn);
     }
 
-    protected void entityInit()
-    {
+    protected void entityInit() {
         super.entityInit();
-        this.dataWatcher.addObject(16, new Byte((byte)0));
+        this.dataWatcher.addObject(16, new Byte((byte) 0));
     }
 
-    public void onUpdate()
-    {
+    public void onUpdate() {
         super.onUpdate();
 
-        if (!this.worldObj.isRemote)
-        {
+        if (!this.worldObj.isRemote) {
             this.setBesideClimbableBlock(this.isCollidedHorizontally);
         }
     }
 
-    protected void applyEntityAttributes()
-    {
+    protected void applyEntityAttributes() {
         super.applyEntityAttributes();
         this.getEntityAttribute(SharedMonsterAttributes.maxHealth).setBaseValue(16.0D);
         this.getEntityAttribute(SharedMonsterAttributes.movementSpeed).setBaseValue(0.30000001192092896D);
     }
 
-    protected String getLivingSound()
-    {
+    protected String getLivingSound() {
         return "mob.spider.say";
     }
 
-    protected String getHurtSound()
-    {
+    protected String getHurtSound() {
         return "mob.spider.say";
     }
 
-    protected String getDeathSound()
-    {
+    protected String getDeathSound() {
         return "mob.spider.death";
     }
 
-    protected void playStepSound(BlockPos p_180429_1_, Block p_180429_2_)
-    {
+    protected void playStepSound(BlockPos p_180429_1_, Block p_180429_2_) {
         this.playSound("mob.spider.step", 0.15F, 1.0F);
     }
 
-    protected Item getDropItem()
-    {
+    protected Item getDropItem() {
         return Items.string;
     }
 
-    protected void dropFewItems(boolean p_70628_1_, int p_70628_2_)
-    {
+    protected void dropFewItems(boolean p_70628_1_, int p_70628_2_) {
         super.dropFewItems(p_70628_1_, p_70628_2_);
 
-        if (p_70628_1_ && (this.rand.nextInt(3) == 0 || this.rand.nextInt(1 + p_70628_2_) > 0))
-        {
+        if (p_70628_1_ && (this.rand.nextInt(3) == 0 || this.rand.nextInt(1 + p_70628_2_) > 0)) {
             this.dropItem(Items.spider_eye, 1);
         }
     }
 
-    public boolean isOnLadder()
-    {
+    public boolean isOnLadder() {
         return this.isBesideClimbableBlock();
     }
 
-    public void setInWeb() {}
+    public void setInWeb() {
+    }
 
-    public EnumCreatureAttribute getCreatureAttribute()
-    {
+    public EnumCreatureAttribute getCreatureAttribute() {
         return EnumCreatureAttribute.ARTHROPOD;
     }
 
-    public boolean isPotionApplicable(PotionEffect p_70687_1_)
-    {
+    public boolean isPotionApplicable(PotionEffect p_70687_1_) {
         return p_70687_1_.getPotionID() == Potion.poison.id ? false : super.isPotionApplicable(p_70687_1_);
     }
 
-    public boolean isBesideClimbableBlock()
-    {
+    public boolean isBesideClimbableBlock() {
         return (this.dataWatcher.getWatchableObjectByte(16) & 1) != 0;
     }
 
-    public void setBesideClimbableBlock(boolean p_70839_1_)
-    {
+    public void setBesideClimbableBlock(boolean p_70839_1_) {
         byte b0 = this.dataWatcher.getWatchableObjectByte(16);
 
-        if (p_70839_1_)
-        {
-            b0 = (byte)(b0 | 1);
-        }
-        else
-        {
+        if (p_70839_1_) {
+            b0 = (byte) (b0 | 1);
+        } else {
             b0 &= -2;
         }
 
         this.dataWatcher.updateObject(16, Byte.valueOf(b0));
     }
 
-    public IEntityLivingData func_180482_a(DifficultyInstance p_180482_1_, IEntityLivingData p_180482_2_)
-    {
+    public IEntityLivingData func_180482_a(DifficultyInstance p_180482_1_, IEntityLivingData p_180482_2_) {
         Object p_180482_2_1 = super.func_180482_a(p_180482_1_, p_180482_2_);
 
-        if (this.worldObj.rand.nextInt(100) == 0)
-        {
+        if (this.worldObj.rand.nextInt(100) == 0) {
             EntitySkeleton entityskeleton = new EntitySkeleton(this.worldObj);
             entityskeleton.setLocationAndAngles(this.posX, this.posY, this.posZ, this.rotationYaw, 0.0F);
-            entityskeleton.func_180482_a(p_180482_1_, (IEntityLivingData)null);
-            this.worldObj.spawnEntityInWorld(entityskeleton);
+            entityskeleton.func_180482_a(p_180482_1_, (IEntityLivingData) null);
+            this.worldObj.addEntity(entityskeleton, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.JOCKEY); // CraftBukkit - add SpawnReason);
             entityskeleton.mountEntity(this);
         }
 
-        if (p_180482_2_1 == null)
-        {
+        if (p_180482_2_1 == null) {
             p_180482_2_1 = new EntitySpider.GroupData();
 
-            if (this.worldObj.getDifficulty() == EnumDifficulty.HARD && this.worldObj.rand.nextFloat() < 0.1F * p_180482_1_.getClampedAdditionalDifficulty())
-            {
-                ((EntitySpider.GroupData)p_180482_2_1).func_111104_a(this.worldObj.rand);
+            if (this.worldObj.getDifficulty() == EnumDifficulty.HARD && this.worldObj.rand.nextFloat() < 0.1F * p_180482_1_.getClampedAdditionalDifficulty()) {
+                ((EntitySpider.GroupData) p_180482_2_1).func_111104_a(this.worldObj.rand);
             }
         }
 
-        if (p_180482_2_1 instanceof EntitySpider.GroupData)
-        {
-            int i = ((EntitySpider.GroupData)p_180482_2_1).field_111105_a;
+        if (p_180482_2_1 instanceof EntitySpider.GroupData) {
+            int i = ((EntitySpider.GroupData) p_180482_2_1).field_111105_a;
 
-            if (i > 0 && Potion.potionTypes[i] != null)
-            {
+            if (i > 0 && Potion.potionTypes[i] != null) {
                 this.addPotionEffect(new PotionEffect(i, Integer.MAX_VALUE));
             }
         }
 
-        return (IEntityLivingData)p_180482_2_1;
+        return (IEntityLivingData) p_180482_2_1;
     }
 
-    public float getEyeHeight()
-    {
+    public float getEyeHeight() {
         return 0.65F;
     }
 
-    class AISpiderAttack extends EntityAIAttackOnCollide
-    {
+    class AISpiderAttack extends EntityAIAttackOnCollide {
+
         private static final String __OBFID = "CL_00002197";
 
-        public AISpiderAttack(Class p_i45819_2_)
-        {
+        public AISpiderAttack(Class p_i45819_2_) {
             super(EntitySpider.this, p_i45819_2_, 1.0D, true);
         }
 
-        public boolean continueExecuting()
-        {
+        public boolean continueExecuting() {
             float f = this.attacker.getBrightness(1.0F);
 
-            if (f >= 0.5F && this.attacker.getRNG().nextInt(100) == 0)
-            {
-                this.attacker.setAttackTarget((EntityLivingBase)null);
+            if (f >= 0.5F && this.attacker.getRNG().nextInt(100) == 0) {
+                this.attacker.setAttackTarget((EntityLivingBase) null);
                 return false;
-            }
-            else
-            {
+            } else {
                 return super.continueExecuting();
             }
         }
 
-        protected double func_179512_a(EntityLivingBase p_179512_1_)
-        {
-            return (double)(4.0F + p_179512_1_.width);
+        protected double func_179512_a(EntityLivingBase p_179512_1_) {
+            return (double) (4.0F + p_179512_1_.width);
         }
     }
 
-    class AISpiderTarget extends EntityAINearestAttackableTarget
-    {
+    class AISpiderTarget extends EntityAINearestAttackableTarget {
+
         private static final String __OBFID = "CL_00002196";
 
-        public AISpiderTarget(Class p_i45818_2_)
-        {
+        public AISpiderTarget(Class p_i45818_2_) {
             super(EntitySpider.this, p_i45818_2_, true);
         }
 
-        public boolean shouldExecute()
-        {
+        public boolean shouldExecute() {
             float f = this.taskOwner.getBrightness(1.0F);
             return f >= 0.5F ? false : super.shouldExecute();
         }
     }
 
-    public static class GroupData implements IEntityLivingData
-        {
-            public int field_111105_a;
-            private static final String __OBFID = "CL_00001700";
+    public static class GroupData implements IEntityLivingData {
 
-            public void func_111104_a(Random p_111104_1_)
-            {
-                int i = p_111104_1_.nextInt(5);
+        public int field_111105_a;
+        private static final String __OBFID = "CL_00001700";
 
-                if (i <= 1)
-                {
-                    this.field_111105_a = Potion.moveSpeed.id;
-                }
-                else if (i <= 2)
-                {
-                    this.field_111105_a = Potion.damageBoost.id;
-                }
-                else if (i <= 3)
-                {
-                    this.field_111105_a = Potion.regeneration.id;
-                }
-                else if (i <= 4)
-                {
-                    this.field_111105_a = Potion.invisibility.id;
-                }
+        public void func_111104_a(Random p_111104_1_) {
+            int i = p_111104_1_.nextInt(5);
+
+            if (i <= 1) {
+                this.field_111105_a = Potion.moveSpeed.id;
+            } else if (i <= 2) {
+                this.field_111105_a = Potion.damageBoost.id;
+            } else if (i <= 3) {
+                this.field_111105_a = Potion.regeneration.id;
+            } else if (i <= 4) {
+                this.field_111105_a = Potion.invisibility.id;
             }
         }
-}
+    }
+}
